### DISCUSSION TITLE
Certificate Not Working on Android Device

### DISCUSSION THREAD

**Discussion by vs0055** [Barre/privaxy#49] [Q&A]
Hi @Barre 

I have been able to run privaxy from source locally on my Mac
and made it available(changing let ip = [0, 0, 0, 0] ) on my home network and connected a android device using proxy on wifi
but the certificate is causing an issue even though I installed certificate on my android device.
it shows following error

Your Connection is not private

NET:ERR_CERT_DATE_INVALID.

I have no problem browsing on my Mac if connected to same proxy and it is effectively blocking all ads.
I want to have this on entire home network so that I don't have to install on all devices.

In application logs I am seeing

Unable to perform handshake for host: go-updater.brave.com:443. Consider excluding it from blocking. The service may not tolerate TLS interception.

Any help here is appreciated.

Thanks

**Comment by Barre** [comment]
Hi,

> NET:ERR_CERT_DATE_INVALID.

Is your device date setup correctly?
May you provide a screenshot of certificate details? Such as:

![image](https://user-images.githubusercontent.com/45085843/210271139-6f7421df-d20a-4ff0-bae8-108d71f4c3f2.png)

https://www.ssl2buy.com/wiki/how-to-view-ssl-certificate-details-in-chrome-firefox-ie-microsoft-edge-vivaldi

**Comment by vs0055** [comment]
Hi @Barre 

here are screenshot of errors


![image](https://user-images.githubusercontent.com/32503303/210397146-8ac41fbe-2a3a-4cde-a025-83be8282ee1f.jpeg)


![image](https://user-images.githubusercontent.com/32503303/210397241-cd09feb3-0f83-425a-925f-587779b81f7c.jpeg)

![image](https://user-images.githubusercontent.com/32503303/210397387-43521f21-8e6e-44b1-8292-6a64695c59f6.jpeg)

**Comment by Karmylr** [comment]
need system trust

**Comment by vs0055** [comment]
Hi @Karmylr @Barre 
Thank you for reply I tried that
but seems like we cannot trust self signed certificate on android (https://stackoverflow.com/questions/37281958/how-to-trust-self-signed-certificate-on-android)
any thoughts on this?

**Comment by leshniak** [comment]
I've had this issue while sharing privaxy from another machine. The cause was that the system clocks drifts a little and the browser on second machine could not validate the certificate. Here's my solution: [privaxy/src/server/cert.rs](https://github.com/leshniak/privaxy/commit/04db4de5c2bfe98a6c43dedde360cf4a15e5ff46#diff-b22a7d5a34c5fef6774649f786af6da2040776b036990d9dc8f94f5bafa5a98a).
